From 9b1123fc772186c826d8bb6411caba0bd54d18c7 Mon Sep 17 00:00:00 2001
From: Roosembert Palacios <roosembert.palacios@epfl.ch>
Date: Sun, 22 Dec 2019 20:26:35 +0100
Subject: [PATCH 4/6] login.c: remove useless indentation on empty lines

Signed-off-by: Roosembert Palacios <roosembert.palacios@epfl.ch>
---
 src/login.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/login.c b/src/login.c
index cc7c797..fa1d2fc 100644
--- a/src/login.c
+++ b/src/login.c
@@ -526,7 +526,7 @@ void auth(
 	if (pwd->pw_shell[0] == '\0')
 	{
 		setusershell();
-		
+
 		char* shell = getusershell();
 
 		if (shell != NULL)
@@ -547,7 +547,7 @@ void auth(
 
 	if (pid == 0)
 	{
-		// set user info 
+		// set user info
 		ok = initgroups(pwd->pw_name, pwd->pw_gid);
 
 		if (ok != 0)
@@ -652,21 +652,21 @@ void auth(
 
 	// close pam session
 	ok = pam_do(pam_close_session, handle, 0, buf);
-	
+
 	if (ok != PAM_SUCCESS)
 	{
 		return;
 	}
 
 	ok = pam_do(pam_setcred, handle, PAM_DELETE_CRED, buf);
-	
+
 	if (ok != PAM_SUCCESS)
 	{
 		return;
 	}
 
 	ok = pam_end(handle, 0);
-	
+
 	if (ok != PAM_SUCCESS)
 	{
 		pam_diagnose(ok, buf);
-- 
2.23.1

