From 520a2097fd20e2330683841fa7429311b4a44317 Mon Sep 17 00:00:00 2001
From: Roosembert Palacios <roosembert.palacios@epfl.ch>
Date: Sun, 27 Oct 2019 17:20:57 +0100
Subject: [PATCH 6/6] login.c: Do not clear the environment upon init

Signed-off-by: Roosembert Palacios <roosembert.palacios@epfl.ch>
---
 src/login.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/login.c b/src/login.c
index ace4e68..ea89172 100644
--- a/src/login.c
+++ b/src/login.c
@@ -207,17 +207,9 @@ void pam_diagnose(int error, struct term_buf* buf)
 
 void env_init(struct passwd* pwd, const char* display_name)
 {
-	extern char** environ;
 	char* term = getenv("TERM");
-	char* lang = getenv("LANG");
-	// clean env
-	environ[0] = NULL;
 
-	if (term != NULL)
-	{
-		setenv("TERM", term, 1);
-	}
-	else
+	if (term == NULL)
 	{
 		setenv("TERM", "linux", 1);
 	}
@@ -228,7 +220,6 @@ void env_init(struct passwd* pwd, const char* display_name)
 	setenv("USER", pwd->pw_name, 1);
 	setenv("LOGNAME", pwd->pw_name, 1);
 	setenv("DISPLAY", display_name, 1);
-	setenv("LANG", lang, 1);
 
 	// Set PATH if specified in the configuration
 	if (strlen(config.path))
-- 
2.23.1

